src/plugins/procmon/procmon.cpp

```diff
@@ -145,6 +145,39 @@ struct process_visitor_ctx
 
 } // namespace
 
+static char* readcmdline(vmi_instance_t vmi, drakvuf_trap_info_t* info, addr_t addr)
+{
+    char *cmd = NULL;
+    access_context_t ctx2 =
+            {
+                    .translate_mechanism = VMI_TM_PROCESS_DTB,
+                    .dtb = info->regs->cr3,
+                    .addr = addr,
+            };
+    addr_t cmdline_addr = ctx2.addr;
+    uint16_t cmd_len = 0;
+    if(VMI_SUCCESS == vmi_read_16(vmi,&ctx2,&cmd_len))
+    {
+        ctx2.addr = cmdline_addr+8;
```

Comment:
Where `+8` came from? If it's an offset in some structure, you could try to find it out using Rekall profile. If for some reason it's not possible, it would be good to define `8` as a constant with meaningful name.

Reply:
CommandLine is a data of type _UNICODE_STRING.
+8 was used to find the Buffer address.

```
  "_RTL_USER_PROCESS_PARAMETERS": [1024, {
   "CommandLine": [112, ["_UNICODE_STRING", {}]], 
   "ConsoleFlags": [24, ["unsigned long", {}]], 
   "ConsoleHandle": [16, ["Pointer", {
    "target": "Void"
   }]]
```

```
 "_UNICODE_STRING": [16, {
   "Buffer": [8, ["Pointer", {
    "target": "unsigned short"
   }]], 
   "Length": [0, ["unsigned short", {}]], 
   "MaximumLength": [2, ["unsigned short", {}]]
  }]
```